### ISSUE TITLE
Chromium panic: runtime error: slice bounds out of range [:-86]

### ISSUE THREAD

**Issue by tim-tx** [moonD4rk/HackBrowserData#120]
**Describe the bug**
Use `./hack-browser-data -vv` paste result here
```

browser.go:158: debug Chromium find bookmark File Success
browser.go:158: debug Chromium find cookie File Success
browser.go:158: debug Chromium find history File Success
browser.go:158: debug Chromium find download File Success
browser.go:158: debug Chromium find password File Success
browser.go:158: debug Chromium find creditcard File Success
[x]:  Get 940 bookmarks, filename is results/chromium_bookmark.csv 
panic: runtime error: slice bounds out of range [:-86]

goroutine 1 [running]:
hack-browser-data/core/decrypt.PKCS5UnPadding(...)
        /src/HackBrowserData/core/decrypt/decrypt.go:189
hack-browser-data/core/decrypt.aes128CBCDecrypt({0xc000202c48?, 0xf1773c001?, 0x0?}, {0xc0002a0640, 0x10, 0x10}, {0xc00001e583, 0x30, 0x30})
        /src/HackBrowserData/core/decrypt/decrypt.go:182 +0x159
hack-browser-data/core/decrypt.ChromePass({0xc000202c48, 0x10, 0x14}, {0xc00001e580, 0x33, 0x33})
        /src/HackBrowserData/core/decrypt/decrypt_linux.go:9 +0xc5
hack-browser-data/core/data.(*cookies).ChromeParse(0xc000261bf0, {0xc000202c48, 0x10, 0x14})
        /src/HackBrowserData/core/data/parse.go:251 +0x648
hack-browser-data/cmd.Execute.func1(0xc000140000?)
        /src/HackBrowserData/cmd/cmd.go:86 +0x48a
github.com/urfave/cli/v2.(*App).RunContext(0xc00013c000, {0x84d328?, 0xc0000b60e0}, {0xc0000a6000, 0x4, 0x4})
        /.go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:315 +0x9cc
github.com/urfave/cli/v2.(*App).Run(...)
        /.go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:215
hack-browser-data/cmd.Execute()
        /src/HackBrowserData/cmd/cmd.go:115 +0x9d2
main.main()
        /src/HackBrowserData/main.go:8 +0x17
```

**Desktop (please complete the following information):**
 - OS Name: Arch Linux
 - Browser Name: Chromium
 - Browser Version: 100.0.4896.75

**Additional context**

**Comment by carlostandfound** [comment]
Having a very similar error myself, and I don't know how to fix it!

**Describe the bug**
Use ./hack-browser-data -vv paste result here

```
[x]:  Get 2559 bookmarks, filename is results/chrome_bookmark.csv
[x]:  Get 1164 cookies, filename is results/chrome_cookie.csv
[x]:  Get 14025 history, filename is results/chrome_history.csv
[x]:  Get 7443 download history, filename is results/chrome_download.csv
panic: runtime error: slice bounds out of range [:-143]

goroutine 1 [running]:
hack-browser-data/core/decrypt.PKCS5UnPadding(...)
        /Users/(me)/HackBrowserData/core/decrypt/decrypt.go:189
hack-browser-data/core/decrypt.aes128CBCDecrypt({0xc0001aa000?, 0xc000c771f0?, 0x0?}, {0xc000c77200, 0x10, 0x10}, {0xc0002bdeab, 0x10, 0x10})
        /Users/(me)/HackBrowserData/core/decrypt/decrypt.go:182 +0x159
hack-browser-data/core/decrypt.ChromePass({0xc0001aa000, 0x10, 0x14}, {0xc0002bdea8, 0x13, 0x13})
        /Users/(me)/HackBrowserData/core/decrypt/decrypt_darwin.go:9 +0xcb
hack-browser-data/core/data.(*passwords).ChromeParse(0xc0001b7440, {0xc0001aa000, 0x10, 0x14})
        /Users/(me)/HackBrowserData/core/data/parse.go:636 +0x3b9
hack-browser-data/cmd.Execute.func1(0xc000242000?)
        /Users/(me)/HackBrowserData/cmd/cmd.go:86 +0x48a
github.com/urfave/cli/v2.(*App).RunContext(0xc00023e000, {0x43ffa78?, 0xc000124008}, {0xc00011a090, 0x3, 0x3})
        /Users/(me)/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:315 +0x9cc
github.com/urfave/cli/v2.(*App).Run(...)
        /Users/(me)/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:215
hack-browser-data/cmd.Execute()
        /Users/(me)/HackBrowserData/cmd/cmd.go:115 +0x9d2
main.main()
        /Users/(me)/HackBrowserData/main.go:8 +0x17
```
**Desktop (please complete the following information):**

OS Name: macOS Monterey v12.1
Browser Name: Chrome
Browser Version: 100.0.4896.88

**Additional context**
My previous laptop died for all intents and purposes - I had to go to data retrieval specialists to get back most of my stuff as raw files, and I was able to properly restore most of it onto my new laptop, but despite my bookmarks, history, etc. being restored in Chrome, my passwords were not. Thing is, I can see all of them in the Login Data file - I know exactly where they are (/Users/(me)/Library/Application Support/Google/Chrome/Default/Login Data) - but they're all encrypted because my passwords had been stored locally, and not through a Google account. Not being able to access my passwords has been heavily debilitating, and for months now I've been searching for a way to decrypt and access them that actually works on Mac, learning everything I now know about coding on-the-fly (my knowledge is still _very_ limited), and I was banking on this program working. Unfortunately, it seems to successfully retrieve everything _but_ my passwords. I even tried using -p and -k commands to direct it straight to the file I need it to access, but it still gives me exactly the same error, which I don't understand, nor do I have any idea how to fix. Any help would be hugely appreciated, to make a massive understatement.

**Comment by moonD4rk** [comment]
Hi guys, I am trying to reproduce this issuses now, you can use the `dev` branch and test if the bug still exists. I was unable to reproduce the problem on my computer.

how to build
```
$ git clone https://github.com/moonD4rk/HackBrowserData.git

$ cd HackBrowserData/cmd/hack-browser-data

$ git checkout dev

$ CGO_ENABLED=1 go build
```


Here's my chrome version info, you could get version info in you chrome tab with `chrome://version` or `chromium://version`

| Google Chrome   | 100.0.4896.127 (Official Build) (arm64)                      |
| --------------- | ------------------------------------------------------------ |
| Revision        | ff0d0695743e65305d7194f9bd309e5e1c824aa0-refs/branch-heads/4896_88@{#4} |
| OS              | macOS Version 12.3.1 (Build 21E258)                          |
| JavaScript      | V8 10.0.139.15                                               |
| User Agent      | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36 |
| Command Line    | /Applications/Google Chrome.app/Contents/MacOS/Google Chrome --flag-switches-begin --flag-switches-end |
| Executable Path | /Applications/Google Chrome.app/Contents/MacOS/Google Chrome |
| Profile Path    | /Users/testuser/Library/Application Support/Google/Chrome/Default |
| Linker          | lld                                                          |

**Comment by moonD4rk** [comment]
HI, guys. I think this issuse has been solved in version[ v0.4.0](https://github.com/moonD4rk/HackBrowserData/releases), if there are still problems you can reopen.

**Comment by tim-tx** [comment]
Seems like problem solved, thanks.
